indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -499,6 +502,7 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
       log.info("Generated [%d] compaction task specs", totalNumSpecs);
 
       int failCnt = 0;
+      Map<String, TaskReport> completionReports = new HashMap<>();
```

Comment:
If the compaction is being run on several intervals (not very likely but still a possibility), can holding all the task reports in memory potentially cause an OOM exception? Currently, most of the task reports contain only `ingestStatsAndErrors` but they may contain other stuff in the future.

In the future, we should consider writing out the sub-reports in a streaming fashion alongwith the required changes to the `TaskReportFileWriter` API.
For now, we should add a guardrail here so that we don't try to hold too many reports in memory and fail with an OOM.

Reply:
what do you think is a good size to hold then ?